### PR TITLE
Fix assertion failed in rule [kdoc-formatting] #1489

### PR DESCRIPTION
**Pull request by icemachined**
Assertion failed: anchorBefore == null || anchorBefore.getTreeParent() == parent in rule [kdoc-formatting] #1489

### What's done:
- fix for KDOC_WRONG_TAGS_ORDER rule
- warning test "tags should be ordered assertion issue"
- fix test "extra new line with tags ordering should not cause assert"

### CODE REVIEW

**Review by nulls — APPROVED** [review]

File: diktat-rules/src/test/resources/test/paragraph2/kdoc/OrderedTagsAssertionTest.kt [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package com.saveourtool.save.reporter.json
+
+/**
+ * Reporter that produces a JSON report as a [Report]
+ *
+ * @property out a sink for output
+ *
```
Comment (akuleshov7) [inline diktat-rules/src/test/resources/test/paragraph2/kdoc/OrderedTagsAssertionTest.kt:7]: we only fix this line here?
Which rule caused an issue?

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocFormatting.kt [hunk 2]
```diff
@@ -219,7 +219,6 @@ class KdocFormatting(configRules: List<RulesConfig>) : DiktatRule(
                     .first()
                     .node
                     .startOffset, basicTags.first().node) {
-                val kdocSection = node.getFirstChildWithType(KDOC_SECTION)!!
```
Comment (akuleshov7) [inline diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocFormatting.kt:222]: I still do not understand why this caused an issue - why this node did not have ChildWithType(KDOC_SECTION)

**Review by 0x6675636b796f75676974687562 — APPROVED** [review]

File: diktat-rules/src/test/resources/test/paragraph2/kdoc/OrderedTagsAssertionTest.kt (hunk 1 above)
Comment (icemachined) [reply diktat-rules/src/test/resources/test/paragraph2/kdoc/OrderedTagsAssertionTest.kt:7]: KDOC_WRONG_TAGS_ORDER property and param in a wrong order + new line between them

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocFormatting.kt (hunk 2 above)
Comment (icemachined) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocFormatting.kt:222]: there are 2 KDOC_SECTION s in this kdoc
1. with description
2. with tags
we get the first one and try to replace tag @property which is in another section.
assert check that anchor should be the child of current section.
so it is better and more universally just to get parent of the replacing tag

**Comment by icemachined** [comment]
Assertion failed: anchorBefore == null || anchorBefore.getTreeParent() == parent in rule [kdoc-formatting] https://github.com/saveourtool/diktat/issues/1489

What's done:
- fix for KDOC_WRONG_TAGS_ORDER rule
- warning test "tags should be ordered assertion issue"
- fix test "extra new line with tags ordering should not cause assert"

File: diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocFormatting.kt (hunk 2 above)
Comment (akuleshov7) [reply diktat-rules/src/main/kotlin/org/cqfn/diktat/ruleset/rules/chapter2/kdoc/KdocFormatting.kt:222]: ah, got it thanks!

**Review by akuleshov7 — APPROVED** [review]

**Pull request merged**